sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,58 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([fields/0]).
+
+fields() ->
+    [ {"foo.setting", fun setting/1}
+    , {"foo.endpoint", fun endpoint/1}
+    , {"foo.greet", fun greet/1}
+    ].
+
+setting(mapping) -> "app_foo.setting";
+setting(type) -> string();
+setting(converter) -> fun to_string/1;
+setting(override_env) -> "MY_OVERRIDE";
+setting(_) -> undefined.
+
+endpoint(mapping) -> "app_foo.endpoint";
+endpoint(type) -> ip();
+endpoint(converter) -> fun to_ip/1;
```

Comment:
`ip()` type should have a builtin converter so we do not have to write it here.
if this is only to demo how converter works, please add a comment.

Reply:
we can have a built-in converter by https://github.com/k32/typerefl/pull/13 .

Reply:
Tag `0.3.0` should support custom validators and parser. Example annotations: https://github.com/k32/typerefl/blob/master/test/transform_tests.erl#L22